### PR TITLE
Produce cleaner JavaDoc and Jacoco output

### PR DESCRIPTION
**Pull request by ejona86**
Instead of producing output for all projects, just do it for projects
that matter to our users.

FYI, @jayantkolhe

### CODE REVIEW

File: all/build.gradle [hunk 1]
```diff
@@ -3,21 +3,23 @@ description = "gRPC: All"
...
+  project(':grpc-okhttp'),
+  project(':grpc-protobuf'),
+  project(':grpc-protobuf-nano'),
+  project(':grpc-stub'),
+]
+
+for (subproject in subprojects) {
     evaluationDependsOn(subproject.path)
 }
+evaluationDependsOn(project(':grpc-interop-testing').path)
```
Comment (zhangkun83) [inline all/build.gradle]: `def jacocoSourceProjects = subprojects + project(':grpc-interop-testing')` ?

Comment (ejona86) [inline all/build.gradle]: I considered something like that, but looping over `jacocoSourceProjects` to call `evaluationDependsOn` is misleading.

Comment (ejona86) [inline all/build.gradle]: Actually, how do you feel about the `evaluationDependsOn` still looping over all of `rootProject.subprojects`, but doing `project(':grpc-interop-testing')` like it is now in the `jacocoMerge`?

Comment (zhangkun83) [inline all/build.gradle]: SGTM

**Comment by ejona86** [comment]
@zhangkun83 PTAL

**Comment by zhangkun83** [comment]
LGTM

**Pull request merged**